src/App.tsx

```diff
@@ -1,61 +1,181 @@
+import React from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
-
-export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/users" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
-    </div>
-  );
-};
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+const todosWithUser = [...todosFromServer].map(todo => ({
+  ...todo,
+  user: usersFromServer.find(
+    (user) => todo.userId === user.id,
+  ) || usersFromServer[0],
+}));
+
+export class App extends React.Component {
+  state = {
+    todos: todosWithUser,
+    users: usersFromServer,
+    validationTitle: true,
+    title: '',
+    validationUser: true,
+    user: '0',
+  };
+
+  handleChange = (event: { target: { name: string; value: string; }; }) => {
+    const { name, value } = event.target;
+
+    if (name === 'title') {
+      this.setState({ validationTitle: true });
+    }
+
+    if (name === 'user') {
+      this.setState({ validationUser: true });
+    }
+
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  addNewTodos = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
+
+    const {
+      title, user, todos, users,
+    } = this.state;
+
+    if (title === '' && user === '0') {
```

Comment:
create a variable with the trimmed title above, and check the trimmed value to avoid saving todo with only spaces in the title